app/services/hyrax/workflow/workflow_importer.rb

```diff
@@ -60,25 +74,34 @@ def validate!
 
       public
 
+      attr_accessor :errors
+
       def call
+        self.errors = []
         Array.wrap(data.fetch(:workflows)).map do |configuration|
-          find_or_create_from(configuration: configuration)
+          begin
+            find_or_create_from(configuration: configuration)
+          rescue InvalidStateRemovalException => e
+            error = I18n.t('hyrax.workflow.load.state_error', workflow_name: e.state.workflow.name, state_name: e.state.name, entity_count: e.state.entities.count)
+            Rails.logger.error(error)
+            errors << error
+            Sipity::Workflow.find_by(name: configuration[:name])
+          end
         end
       end
 
       private
 
         def find_or_create_from(configuration:)
-          workflow = Sipity::Workflow.find_or_initialize_by(name: configuration.fetch(:name)) do |wf|
```

Comment:
The reorganization of this method does not look as though it changes anything. Why was it reorganized?

I'm concerned that we would be interacting with a non-persisted workflow. The previous version found the workflow, updated attributes, then saved (and threw an exception if invalid). Then it proceeded to generate the state_diagram and find the workflow permission.

The new version, finds or initialized, makes transformative changes on the state_diagram and permissions, then attempts to save the workflow (with validation). I think this could result in inconsistent data.